### PR TITLE
Upstream merge (11/28/24)

### PR DESCRIPTION
**Pull request by PursuitInAshes**
## About the PR
Merges wizden stable up to 11/28/24.
Merges the following CD PRs:

Merges https://github.com/cosmatic-drift-14/cosmatic-drift/pull/467
Merges https://github.com/cosmatic-drift-14/cosmatic-drift/pull/471
Merges https://github.com/cosmatic-drift-14/cosmatic-drift/pull/472
Merges https://github.com/cosmatic-drift-14/cosmatic-drift/pull/477
Merges https://github.com/cosmatic-drift-14/cosmatic-drift/pull/485
Merges https://github.com/cosmatic-drift-14/cosmatic-drift/pull/478
https://github.com/cosmatic-drift-14/cosmatic-drift/pull/484
https://github.com/cosmatic-drift-14/cosmatic-drift/pull/487

Reverts Umbra PR b770e0a

The revert of that commit and merge of 471 is simple code clean up.

### CODE REVIEW

File: Content.Server/StationEvents/Events/IonStormRule.cs
Comment (OctoRocket) [inline Content.Server/StationEvents/Events/IonStormRule.cs:1]: Seems to have partially reverted with the merge

**Review by OctoRocket — CHANGES_REQUESTED** [review]

File: Content.Server/StationEvents/Events/IonStormRule.cs
Comment (OctoRocket) [reply Content.Server/StationEvents/Events/IonStormRule.cs:1]: Misread that, it actually is just using features that are no longer included in the file

**Review by OctoRocket — APPROVED** [review]

**Comment by OctoRocket** [comment]
`Failed!  - Failed:    54, Passed:     5, Skipped:     0, Total:    59, Duration: 4 s - Content.IntegrationTests.dll (net8.0)`

Seems like the tests really like this one

**Comment by OctoRocket** [comment]
The test fails were fixed 2 weeks ago in Robust Toolbox's netserializer, seems like the latest Robust Toolkit version was just not pulled.

**Review by TsjipTsjip — APPROVED** [review]
Works on Coolsurf6's machine

**Comment by Coolsurf6** [comment]
*grins

**Pull request merged**